### ISSUE TITLE
cryptic call to onConnect in example/client.js

### ISSUE THREAD

**Issue by grondilu** [nknorg/nkn-sdk-js#94]
I'm trying to understand example/client.js, and this part confuses me :

https://github.com/nknorg/nkn-sdk-js/blob/f6fa9cdb751a45eaf32e432d2e80bc0ab7efb50f/examples/client.js#L26-L29

What does this do?  It waits for both clients to add a `resolve` function to their event listener stack, but what will that function end up being if the promise is never consumed by a `then`?

Also I can't help to notice that the code works just fine if we remove these four lines.  I suspect the author meant to do something else.

**Comment by yilunzhang** [comment]
This piece of the code is basically just wait until both clients trigger `onConnect` event (i.e. connected to a node) before executing the rest of the examples (using `await` since the whole example is in an `async` function).

Right after this part there is another await that waits for 1s. If you remove both of them, you should see an error when trying to send message:
```
ClientNotReadyError: client not ready
```

That's what happens when trying to send message before client is connected.

**Comment by grondilu** [comment]
> Right after this part there is another await that waits for 1s.

Well, precisely.  Why do you have to do that if the preceding code was meant to wait for both clients to be ready?

**Comment by yilunzhang** [comment]
By default multiclient is used in examples. A multiclient has multiple clients. The onConnect event of a multiclient will be triggered if ANY client is connected. We added that 1s delay so more clients can connect (which leads to better performance and reliability). It's not necessary to do that, and 1s is just some arbitrary value we used in example.

In practice you typically don't need to do that because clients are usually long lasting and created when app is started, while sending messages are triggered by events, so there are natural delays usually.
